### DISCUSSION TITLE
Image | NanoPi R2C Plus

### DISCUSSION THREAD

**Discussion by MichaIng** [MichaIng/DietPi#6035] [Image Request]
Dedicated image required as of new Ethernet chip and different required device tree.

Related request on the forum: https://dietpi.com/forum/t/support-for-nanopi-r2c/6369

**Comment by TheLinuxGuy** [comment]
+1 
I have this device and support would be great.

**Comment by baffian** [comment]
+1

**Comment by MichaIng** [comment]
Converted to discussion, so you can upvote now the OP.

**Comment by Hairo** [comment]
Any news about this? apparently the device tree was added in mainline kernel 6.5.
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d211665c5a833873ee37e501af58adbf028e6b5f

**Reply by MichaIng** [reply]
Can you flash our NanoPi R2S image, and then test the following:
```sh
ip l # verify that only one Ethernet interface is visible
G_CONFIG_INJECT 'fdtfile=' 'fdtfile=rockchip/rk3328-nanopi-r2-rev06.dtb' /boot/dietpiEnv.txt
reboot
ip l # after reboot, check whether there are now two Ethernet interfaces visible
```

**Reply by Hairo** [reply]
I tried with the R2S bookworm image and it seems to be booting (SD card is expanded) but network connection doesn't seem to be working, tried both ports, with a static and DHCP IP, and using that dietpiEnv.txt setting, nothing worked. Can't ping or ssh to the static IP and it doesn't show up when scanning the network.

**Reply by MichaIng** [reply]
Does `ip l` show any Ethernet interface? If not, are there any kernel errors?
```sh
dmesg -l 0,1,2,3
```
Ah, I guess you have no UART adapter to check the serial console, do you? I'll check whether we can enable the USB serial module, so you should be able to power the SBC from another computer via USB and connect to the serial console this way as well.

The device tree adds support for the eMMC (Plus vs non-Plus) only, while hence the Ethernet adapter should work with the changed fdtfile in `dietpiEnv.txt`. Did you edit it (on the DIETPISETUP FAT partition) before booting the SBC the first time?
